### ISSUE TITLE
Replicate first paper results

### ISSUE THREAD

**Issue by stevehadd** [Fracappo87/XBTs_classification#8]
As a starting point for results for our paper on the XBT ML problem, we should try to replicate the results in the first paper on this problem by members Lab summer placements Thomas P. Leahy and Francesc Pons Llopis, together with Matt Pallmer  and Niall Robinson. 

The paper can be found here:
https://journals.ametsoc.org/doi/full/10.1175/JTECH-D-18-0012.1

Implemented in code located here:
https://github.com/ThomasLeahy/NN_XBT/blob/master/NN_XBT.py

The details of the ML setup used, taken from the paper, are:
* train and test for each year independently
* input variables (same as iMeta)
  * country
  * max depth
  * profile date
* multi layer NN 
  * 10 nodes/layer, 
  * 2 hidden layers 
  * relu activation function
  * softmax loss function
  * scikit learn NN implementation
* hyerparameters (based on https://arxiv.org/abs/1412.6980, default settings in scikit learn):
  * initial learning rate 0.001
  * exponential decay rate for estimates of first moment vector 0.9
  * exponential decay rate for estimates of second moment vector 0.999
  * value for numerical stability 10^-8
  * L2 penalty parameter (regularization term) 0.0001
  * tolerance 10^-4
* input encoding:
  * z-score normalisation
  * ordinal encoding for categorical values (e.g. country)
* 75/25 % train/test split
* training: adam for gradient descent
* performance metric: recall (also known as hit rate, also used for iMeta paper)

**Comment by stevehadd** [comment]
The results have now been replicated, so closing the ticket.
